CONTRIBUTING

```diff
@@ -15,15 +15,14 @@ https://twistedmatrix.com/trac/wiki/ContributingToTwistedLabs
 
 Twisted has a Code of Conduct, available at code_of_conduct.md.
 
-**Warning: pull requests are ignored** unless they have an associated ticket in trac.
```

Comment:
FWIW -- currently the only way to kick the auto-test machinery is to open a pull request, so it would be nice if there was a way to say "this is not ready for review yet". Maybe we can suggest an explicit "WIP" or "In-progress" tag in the template for such things?

Reply:
I know the pain.

Not sure what is a "tag" :)

Labels will not work as only contributors can change them.

---------

For now, things should be "simple". Set the "review" keyword on the Trac ticket. This will show in the review report https://twistedmatrix.com/trac/report/25

--------

What I am doing for other repos, is using a "needs-review" keyword in a comment.

There is a webhook listening for comments, and when needs-review is there it auto updates Trac and can do other things... like setting a label.

Reply:
Usually I've seen this done by a prefix on the PR (e.g., "WIP:" means "work in progress"). If we all agree that nobody reviews until it's marked "review" in trac that's fine -- we should make it explicit, for contributors and reviewers alike. Nobody likes their half-done work judged :)

Reply:
The Trac keyword thing is described at 

http://twistedmatrix.com/trac/wiki/TwistedDevelopment#SubmittingaPatch 

and this link is part of the new template.

----------

I saw the WIP in the title, but I don't like it.

You request once a review... but then changes are required, should the reviewer set WIP back to the title and once is ready for re-review the author remove the WIP title?

For me, the 'needs-review' work best, as once all is ready for re-review, you will leave a comment addresing what whas changed and at the end you leave the keyword.

No need to scroll to the top, click edit button and change the title :)